### ISSUE TITLE
Warn about driver CUDA mismatch in torch.cuda.is_available()

### ISSUE THREAD

**Issue by colesbury** [pytorch/pytorch#30528]
## 🐛 Enhancement

The `torch.cuda.is_available()` should print a warning if the GPU driver is too old to support the CUDA version installed with PyTorch. If the system does not have a GPU or PyTorch was built without CUDA support, the function should return False but remain silent to avoid spamming less useful messages.

We already print such a warning about driver mismatch when initializing CUDA. However, I've seen a number of shared PyTorch scripts that don't even get to initializing CUDA because its gated on a `torch.cuda.is_available()` call. People are usually a little lost at that point, because they don't know **why** `torch.cuda.is_available()` returned `False`.

**Comment by albanD** [comment]
Should we then consider adding a new function that will print out why `cuda.is_available()` is `False`?

**Comment by bobvo23** [comment]
Yes, being able to set a verbose parameter would be helpful to resolve the missing CUDA problem.

**Comment by epignatelli** [comment]
Having the same problem here.

The main use case is to detect not if cuda is actually available, but if cuda can be used for computation.

Maybe a function `cuda.is_supported()` might do?
